rpcs3/Crypto/unself.cpp

```diff
@@ -1293,9 +1293,43 @@ fs::file SELFDecrypter::MakeElf(bool isElf32)
 			// PHDR type.
 			if (meta_shdr[i].type == 2)
 			{
-				// Seek to the program header data offset and write the data.
-				e.seek(phdr32_arr[meta_shdr[i].program_idx].p_offset);
-				e.write(data_buf.get() + data_buf_offset, meta_shdr[i].data_size);
+				// Decompress if necessary.
+				if (meta_shdr[i].compressed == 2)
+				{
+					// Store the length in writeable memory space.
+					std::unique_ptr<uLongf> decomp_buf_length(new uLongf);
+					memcpy(decomp_buf_length.get(), &phdr32_arr[meta_shdr[i].program_idx].p_filesz, sizeof(uLongf));
+
+					/// Create a pointer to a buffer for decompression.
+					std::unique_ptr<u8[]> decomp_buf(new u8[phdr32_arr[meta_shdr[i].program_idx].p_filesz]);
```

Comment:
Why not use `decomp_buf_length` instead of `phdr32_arr[meta_shdr[i].program_idx].p_filesz`?